lambda/src/main/scala/pricemigrationengine/model/AmendmentData.scala

```diff
@@ -63,18 +67,54 @@ object AmendmentData {
       chargeFromProduct: ZuoraProductRatePlanCharge
   )
 
+  def priceData(
+      account: ZuoraAccount,
+      catalogue: ZuoraProductCatalogue,
+      subscription: ZuoraSubscription,
+      invoiceList: ZuoraInvoiceList,
+      nextServiceStartDate: LocalDate,
+      cohortSpec: CohortSpec,
+  ): Either[AmendmentDataFailure, PriceData] = {
+
+    // Note: The nextServiceDate has gone through:
+    // cohortSpec.earliestPriceMigrationStartDate >> `spreadEarliestStartDate` >> `nextServiceStartDate`
+
+    /*
+      Date: March 2023
+      Author: Pascal
```

Comment:
Isn't this going to be clear from the history?

Reply:
In many projects I tend to date and sign important comments, above all if they might come across as a surprise to people who look at the code and without having seen the PR. Maybe not as important in this project actually. In any case, I will probably remove the entire section when I come back doing more refactoring :)